org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java

```diff
@@ -106,13 +108,17 @@ public RefactoringStatus checkInitialConditions(IProgressMonitor pm)
 			params.setTextDocument(identifier);
 			params.setPosition(LSPEclipseUtils.toPosition(offset, document));
 
+			@SuppressWarnings("null")
+			List<Pair<LanguageServerWrapper, Either3<Range, PrepareRenameResult, PrepareRenameDefaultBehavior>>> list = LanguageServers
+					.forDocument(document).withFilter(LSPRenameProcessor::isPrepareRenameProvider)
+					.collectAll((w, ls) -> ls.getTextDocumentService().prepareRename(params)
+							.thenApply(result -> new Pair<>(w, result)))
+					.get(1000, TimeUnit.MILLISECONDS);
 
-			Optional<Pair<LanguageServerWrapper, Either3<Range, PrepareRenameResult, PrepareRenameDefaultBehavior>>> tmp
-				= LanguageServers.forDocument(document).withFilter(LSPRenameProcessor::isPrepareRenameProvider)
-				.computeFirst((w, ls) -> ls.getTextDocumentService().prepareRename(params).thenApply(result -> new Pair<>(w, result)))
-				.get(1000, TimeUnit.MILLISECONDS);
+			Optional<Pair<LanguageServerWrapper, Either3<Range, PrepareRenameResult, PrepareRenameDefaultBehavior>>> tmp = list
+					.stream().filter(Objects::nonNull).filter(t -> t.getSecond() != null).findFirst();

```

Comment:
OK, I see the `getSecond` just match the condition that used to be below, after `tmp.isEmpty()`, so this code is indeed an improvement over current one.
I'm still unsure about only checking `getSecond`, but it's a different issue I guess.